sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs

```diff
@@ -1314,13 +1437,58 @@ private void SetNameFieldsIfNull()
                     scope.Start();
                     ResponseWithHeaders<FileStartCopyHeaders> response;
 
+                    if (copySourceFileCreatedOn == true && smbProperties?.FileCreatedOn != null)
+                    {
+                        throw new ArgumentException($"{nameof(copySourceFileCreatedOn)} and {nameof(FileSmbProperties.FileCreatedOn)} cannot both be set.");
```

Comment:
Validating these is great, but we should be more specific with the names we're giving people.  `copySourceFileCreatedOn` is an internal only property and instead we should say something more like 

```suggestion
                        throw new ArgumentException($"{nameof(ShareFileCopyOptions)}.{nameof(ShareFileCopyOptions.CopySourceFileCreatedOn)} and {nameof(ShareFileCopyOptions)}.{nameof(FileSmbProperties)}.{nameof(FileSmbProperties.FileCreatedOn)} cannot both be set.");
```

Reply:
Fixed.